pkg/services/authn/authnimpl/service.go

```diff
@@ -353,6 +353,7 @@ func (s *Service) resolveIdenity(ctx context.Context, orgID int64, namespaceID a
 			ID:    namespaceID.String(),
 			OrgID: orgID,
 			ClientParams: authn.ClientParams{
+				AllowGlobalOrg:  true,
```

Comment:
Are we suppose to allow service account to use gloabl org?

Reply:
They shouldn't be able to do anything in the global org, so any checks in the global org should fail. What happens now though is that `resolveIdenity` sets user's org ID to their current org (because `AllowGlobalOrg` is set to `false`), so we evaluate permissions in the current org instead of the global org, and the evaluation passes.

Maybe a more accurate name for the parameter would be something like `AllowResolvingGlobalIdentity` (this is too long, but something on those lines).